batch/sql/estimated-current.sql

```diff
@@ -1306,6 +1338,85 @@ BEGIN
   COMMIT;
 END $$
 
+DROP PROCEDURE IF EXISTS cancel_job_group $$
```

Comment:
Any reason to not delete the procedure `cancel_batch`? I would appreciate being able to see the git diff between `cancel_batch` and `cancel_job_group`.

Reply:
Unfortunately, we can't delete `cancel_batch` until the new batch code is deployed that uses `cancel_job_group`.

Reply:
```diff
1,3c1,4
< DROP PROCEDURE IF EXISTS cancel_batch $$
< CREATE PROCEDURE cancel_batch(
<   IN in_batch_id VARCHAR(100)
---
> DROP PROCEDURE IF EXISTS cancel_job_group $$
> CREATE PROCEDURE cancel_job_group(
>   IN in_batch_id VARCHAR(100),
>   IN in_job_group_id INT
23c24
<                               WHERE id = in_batch_id
---
>                               WHERE id = in_batch_id AND job_group_id = in_job_group_id
45c46,48
<     WHERE job_group_inst_coll_cancellable_resources.batch_id = in_batch_id AND batch_updates.committed
---
>     WHERE job_group_inst_coll_cancellable_resources.batch_id = in_batch_id AND
>       job_group_inst_coll_cancellable_resources.job_group_id = in_job_group_id AND
>       batch_updates.committed
57,58c60,68
<     # there are no cancellable jobs left, they have been cancelled
<     DELETE FROM job_group_inst_coll_cancellable_resources WHERE batch_id = in_batch_id;
---
>     # delete all rows that are children of this job group
>     DELETE job_group_inst_coll_cancellable_resources FROM job_group_inst_coll_cancellable_resources
>     LEFT JOIN batch_updates ON job_group_inst_coll_cancellable_resources.batch_id = batch_updates.batch_id AND
>       job_group_inst_coll_cancellable_resources.update_id = batch_updates.update_id
>     INNER JOIN job_group_self_and_ancestors ON job_group_inst_coll_cancellable_resources.batch_id = job_group_self_and_ancestors.batch_id AND
>       job_group_inst_coll_cancellable_resources.job_group_id = job_group_self_and_ancestors.job_group_id
>     WHERE job_group_inst_coll_cancellable_resources.batch_id = in_batch_id AND
>       job_group_self_and_ancestors.ancestor_id = in_job_group_id AND
>       batch_updates.committed;
60,61c70,74
<     # cancel root job group only
<     INSERT INTO job_groups_cancelled (id, job_group_id) VALUES (in_batch_id, 0);
---
>     INSERT INTO job_groups_cancelled
>     SELECT batch_id, job_group_id
>     FROM job_group_self_and_ancestors
>     WHERE batch_id = in_batch_id AND ancestor_id = in_job_group_id
>     ON DUPLICATE KEY UPDATE job_group_id = job_groups_cancelled.job_group_id;
```

Reply:
Hmm. OK, you're right. Let's make sure we immediately follow up with a PR to delete cancel_batch so we don't forget to do that.